### DISCUSSION TITLE
Why typia.json.application() does not support patternProperties?

### DISCUSSION THREAD

**Discussion by miyaji255** [samchon/typia#1254] [Q&A]
Why typia.json.application() does not support patternProperties?
OAS 3.0 does not support pertternProperties, but OAS 3.1 supports.

https://github.com/OAI/OpenAPI-Specification/issues/687

**Comment by samchon** [comment]
OpenAPI 3.1 / JSON Schema supports `patternProperties`, but typia cannot infer it automatically from `Record<string, T>` or a TypeScript index signature because TypeScript does not retain an arbitrary regex key constraint there.

For schema-only consumers, the supported escape hatch is `tags.JsonSchemaPlugin` / custom `TagBase.schema`. I opened #2002 to document this pattern, including the caveat that `JsonSchemaPlugin` does not change runtime validator behavior. If runtime validation must reject non-matching keys, that rule needs a validating custom tag or separate key validation.

I will leave this discussion open until the documentation PR is merged.
